pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/GeoPersistentReplicator.java

```diff
@@ -184,12 +183,20 @@ protected boolean replicateEntries(List<Entry> entries) {
                     msg.setSchemaInfoForReplicator(schemaFuture.get());
                     msg.getMessageBuilder().clearTxnidMostBits();
                     msg.getMessageBuilder().clearTxnidLeastBits();
+                    // Increment pending messages for messages produced locally
+                    PENDING_MESSAGES_UPDATER.incrementAndGet(this);
+                    shouldRollbackPendingSendAdd.setTrue();
                     producer.sendAsync(msg, ProducerSendCallback.create(this, entry, msg));
```

Comment:
Why do we need `shouldRollbackPendingSendAdd`? Is `producer.sendAsync(msg, ProducerSendCallback.create(this, entry, msg))` throws exception?

Reply:
Already removed these lines, please take a look again